src/daemon/rpmostreed-transaction-types.c

```diff
@@ -736,44 +731,36 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
 
   if (self->packages_added)
     {
-      g_autoptr(GHashTable) pkgs =
-        g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
-      g_autoptr(GHashTable) local_pkgs =
-        g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+      if (!rpmostree_origin_add_packages (origin, self->packages_added,
+                                          FALSE, cancellable, error))
+        goto out;
+    }
 
-      for (char **it = self->packages_added; it && *it; it++)
+  if (self->local_packages_added != NULL)
+    {
+      /* add them all to an array first to make the origin
+       * update more efficient */
+      g_autoptr(GPtrArray) pkgs = g_ptr_array_new_with_free_func (g_free);
+
+      gint nfds = 0;
+      const gint *fds =
```

Comment:
Good catch! I initially used `peek_fds` instead of `steal_fds`, but in the end didn't need to. I switched it back to `peek_fds` so they'll be closed when it gets unref'ed.